klippy/extras/probe.py

```diff
@@ -180,15 +183,31 @@ def get_last_xy_home_positon(self):
             return self.probe.last_home_position()
         else:
             return None
-    def lift_z(self, z_pos):
+    def lift_z(self, z_pos, add=False):
         # Lift toolhead
         curpos = self.toolhead.get_position()
-        curpos[2] = z_pos
+        if add:
+            curpos[2] += z_pos
+        else:
+            curpos[2] = z_pos
         try:
             self.toolhead.move(curpos, self.lift_speed)
         except homing.EndstopError as e:
             self.finalize(False)
             raise self.gcode.error(str(e))
+    def probe_point(self):
+        avg_z = 0.
+        for i in range(self.samples):
+            self.gcode.run_script_from_command("PROBE")
+            self.toolhead.wait_moves()
+            kin_pos = self.callback.get_probed_position()
+            avg_z += kin_pos[2]
```

Comment:
I believe this would break delta calibration.  The delta calibration returns a "stable position" (which is the number of steps taken since hitting the endstop on each tower).  At a high-level I was thinking that get_probed_position() would return a "cookie" that the ProbePointsHelper class would just pass to finalize().

Could the additional samples just be passed to finalize() and let finalize() do the averaging?  (The other users of ProbePointsHelper will do that automatically.)